toolbox/CMakeLists.txt

```diff
@@ -202,57 +221,136 @@ configure_block(BLOCK_NAME "Get Limits"
 get_property(ALL_HEADERS GLOBAL PROPERTY WBT_HEADERS)
 get_property(ALL_SOURCES GLOBAL PROPERTY WBT_SOURCES)
 
-add_definitions(-DMATLAB_MEX_FILE)
-include_directories(include)
-include_directories(include/base)
+add_library(WBToolboxLibrary SHARED
+    ${ALL_HEADERS} ${ALL_SOURCES}
+)
+set_target_properties(WBToolboxLibrary PROPERTIES
+    OUTPUT_NAME "WBToolbox"
+    # VERSION ${PROJECT_VERSION}
+    PUBLIC_HEADER "${ALL_HEADERS}"
+)
+
+# Import math symbols from standard cmath
+target_compile_definitions(WBToolboxLibrary PUBLIC "-D_USE_MATH_DEFINES")
+
+# Set other properties of the target WBToolbox
+# --------------------------------------------
 
-include_directories(SYSTEM ${Matlab_INCLUDE_DIRS} "${Matlab_ROOT_DIR}/simulink/include")
-include_directories(SYSTEM ${YARP_INCLUDE_DIRS} ${EIGEN3_INCLUDE_DIR})
+target_include_directories(WBToolboxLibrary PUBLIC
+    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
+    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include/base>
+)
+
+target_include_directories(WBToolboxLibrary SYSTEM PUBLIC
+    ${Matlab_INCLUDE_DIRS}
+    ${Matlab_ROOT_DIR}/simulink/include
+    ${YARP_INCLUDE_DIRS}
```

Comment:
I think this is not necessary, and this should be supported just by linking YARP .

Reply:
Yes they are not required. Linking YARP is sufficient. By the way, would you recommend to limit the YARP components as I do with iDynTree?